sway-core/src/semantic_analysis/ast_node/expression/match_expression/typed/typed_match_branch.rs

```diff
@@ -130,15 +130,17 @@ impl ty::TyMatchBranch {
             expression: typed_result_expression_variant,
             return_type: typed_result_return_type,
             span: typed_result_span,
-        } = typed_result;
+        } = typed_result.clone();
```

Comment:
To avoid cloning the `typed_result` twice we can replace this deconstructing with a simple:
```Sway
        let typed_result_return_type = typed_result.return_type;
        let typed_result_span = typed_result.span.clone();
        match typed_result.expression {
```